src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function(x) {
+    if (x === undefined) {
+      return 0;
+    }
+    sum += x;
+
+    const adder = function(y) {
+      if (y !== undefined) {
+        sum += y;
+
+        return adder;
+      }
+
+      return sum;
+    };
```

Comment:
you don't need this function. it makes logic too complicated. your first function(function(x)) can be the adder and return itself